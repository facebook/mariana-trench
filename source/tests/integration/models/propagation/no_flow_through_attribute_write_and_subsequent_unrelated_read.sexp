(method (private) "LFlow;.flow:(LData;)V"
 (
  (load-param v0)
  (load-param v1)

  LINE(1)
  (invoke-virtual (v0) "LSource;.source:()LData;")
  (move-result-object v2)

  (iput-object v2 v1 "LData;.field:LData;")

  LINE(2)
  (iget-object v1 "LData;.other_field:LData;")
  (move-result-pseudo-object v3)

  LINE(3)
  (invoke-virtual (v0 v3) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
