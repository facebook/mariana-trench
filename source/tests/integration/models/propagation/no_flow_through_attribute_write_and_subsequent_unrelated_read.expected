{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "method" : "LFlow;.flow:(LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 3,
            "path" : "Flow.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1).other_field",
          "distance" : 1,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    }
  ]
}
